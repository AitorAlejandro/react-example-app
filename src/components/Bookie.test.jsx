const React = require('react');
const ReactDOM = require('react-dom');
import ReactDOMServer from 'react-dom/server';

const TestUtils = require('react-addons-test-utils');
const Bookie = require('./Bookie').default;
const renderer = require('react-test-renderer');

jest.mock('../lib/Scheduler');
jest.mock('../lib/BusyAdapter');

it('should exists', () => {
  const bookie = TestUtils.renderIntoDocument(<Bookie />);
  expect(TestUtils.isCompositeComponent(bookie)).toBeTruthy();
});

// it('markup matches snapshot', () => {
//   const tree = renderer.create(<Bookie />).toJSON();
//   expect(tree).toMatchSnapshot();
// });

fdescribe('enabling bookie button', () => {
  // there's a gap on 2017-01-01 at 17:00-18:00
  const savedState = require('./Bookie.test.data').savedState;

  const createBookieWithState = function (state) {
    const result = TestUtils.renderIntoDocument(<Bookie />);
    result.setState(state);
    result.negotiateStateDiff({}, true);
    return result;
  }

  fit('is enabled for valid range', () => {
    // const bookieComponent = createBookieWithState(savedState);

    const bookieComponent = TestUtils.renderIntoDocument(<Bookie />);
    bookieComponent.setState(require('./Bookie.test.data').savedState);
    bookieComponent.negotiateStateDiff({
      "dayPicked": "2017-01-01",
      "hourPicked": 18,
      "minutesIdxPicked": 1,
    });

    bookieComponent.negotiateStateDiff({
      "endVal": "2017-01-01T19:00:00+02:00"
    });

    // ensure booking is enabled in state
    expect(bookieComponent.state.bookingAllowed).toBe(true);

    // ensure booking button is enabled
    expect(ReactDOM.findDOMNode(bookieComponent).querySelector('button').disabled).toBe(false);

    // check markup snapshot
    const reactElement = bookieComponent.render();
    const renderedMarkup = ReactDOMServer.renderToStaticMarkup(reactElement);
    expect(renderedMarkup).toMatchSnapshot();
  });

  it('is disabled for the range with a gap', () => {
    const state = Object.assign({}, savedState, {
      "hourPicked": 16,
      "minutesIdxPicked": 2,
      "endVal": "2017-01-01T18:15:00+02:00",
    });

    const bookieComponent = createBookieWithState(state);

    expect(bookieComponent.state.bookingAllowed).toBe(false);
    expect(ReactDOM.findDOMNode(bookieComponent).querySelector('button').disabled).toBe(true);

    const reactElement = bookieComponent.render();
    const renderedMarkup = ReactDOMServer.renderToStaticMarkup(reactElement);
    expect(renderedMarkup).toMatchSnapshot();
  });
});