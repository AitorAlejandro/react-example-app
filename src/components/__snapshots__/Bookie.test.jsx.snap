exports[`enabling bookie button is disabled for the range with a gap 1`] = `"<div class=\"bookie-container\"><ul class=\"pick-day\"><li class=\"disabled\"><a>&lt;&lt;</a></li><li class=\"current disabled\"><a>01/01/2017</a></li><li class=\"disabled\"><a>01/02/2017</a></li><li class=\"disabled\"><a>01/03/2017</a></li><li class=\"disabled\"><a>01/04/2017</a></li><li class=\"disabled\"><a>&gt;&gt;</a></li></ul><ul class=\"pick-minutes\"><li class=\"disabled\"><a>00</a></li><li class=\"disabled\"><a>15</a></li><li class=\"current disabled\"><a>30</a></li><li class=\"disabled\"><a>45</a></li></ul><ul class=\"pick-hour\"><li class=\"disabled\"><a>&lt;&lt;</a></li><li class=\"current disabled\"><a>4pm</a></li><li class=\"disabled\"><a>5pm</a></li><li class=\"disabled\"><a>6pm</a></li><li class=\"disabled\"><a>7pm</a></li><li class=\"disabled\"><a>&gt;&gt;</a></li></ul><div class=\"range\"><span class=\"pick start picking\">01-01 4pm:30</span><span> — </span><span class=\"pick end\">01-01 6pm:15</span></div><div class=\"text-center\"><button disabled=\"\" type=\"button\" class=\"btn btn-default\">Book</button></div></div>"`;

exports[`enabling bookie button is enabled for valid range 1`] = `"<div class=\"bookie-container\"><ul class=\"pick-day\"><li class=\"disabled\"><a>&lt;&lt;</a></li><li class=\"current disabled\"><a>01/01/2017</a></li><li class=\"disabled\"><a>01/02/2017</a></li><li class=\"disabled\"><a>01/03/2017</a></li><li class=\"disabled\"><a>01/04/2017</a></li><li class=\"disabled\"><a>&gt;&gt;</a></li></ul><ul class=\"pick-minutes\"><li class=\"disabled\"><a>00</a></li><li class=\"current disabled\"><a>15</a></li><li class=\"disabled\"><a>30</a></li><li class=\"disabled\"><a>45</a></li></ul><ul class=\"pick-hour\"><li class=\"disabled\"><a>&lt;&lt;</a></li><li class=\"disabled\"><a>5pm</a></li><li class=\"current disabled\"><a>6pm</a></li><li class=\"disabled\"><a>7pm</a></li><li class=\"disabled\"><a>8pm</a></li><li class=\"disabled\"><a>&gt;&gt;</a></li></ul><div class=\"range\"><span class=\"pick start picking\">01-01 6pm:15</span><span> — </span><span class=\"pick end\">01-01 7pm:00</span></div><div class=\"text-center\"><button type=\"button\" class=\"btn btn-default\">Book</button></div></div>"`;

exports[`test markup matches snapshot 1`] = `
<div
  className="bookie-container">
  <ul
    className="pick-day">
    <li
      className="disabled"
      onClick={undefined}>
      <a>
        <<
      </a>
    </li>
    <li
      className="current"
      onClick={[Function]}>
      <a>
        01/02/2017
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}>
      <a>
        01/03/2017
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}>
      <a>
        01/04/2017
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}>
      <a>
        01/05/2017
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}>
      <a>
        >>
      </a>
    </li>
  </ul>
  <ul
    className="pick-minutes">
    <li
      className="disabled"
      onClick={undefined}>
      <a>
        00
      </a>
    </li>
    <li
      className="disabled"
      onClick={undefined}>
      <a>
        15
      </a>
    </li>
    <li
      className="disabled"
      onClick={undefined}>
      <a>
        30
      </a>
    </li>
    <li
      className="disabled"
      onClick={undefined}>
      <a>
        45
      </a>
    </li>
  </ul>
  <ul
    className="pick-hour">
    <li
      className="disabled"
      onClick={undefined}>
      <a>
        <<
      </a>
    </li>
    <li
      className="disabled"
      onClick={undefined}>
      <a>
        5pm
      </a>
    </li>
    <li
      className="disabled"
      onClick={undefined}>
      <a>
        6pm
      </a>
    </li>
    <li
      className="disabled"
      onClick={undefined}>
      <a>
        7pm
      </a>
    </li>
    <li
      className="disabled"
      onClick={undefined}>
      <a>
        8pm
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}>
      <a>
        >>
      </a>
    </li>
  </ul>
  <div
    className="range">
    <span
      className="pick start picking"
      onClick={[Function]}>
      choose start
    </span>
    <span>
       — 
    </span>
    <span
      className="pick end"
      onClick={[Function]}>
      choose end
    </span>
  </div>
  <div
    className="text-center">
    <button
      className="btn btn-default"
      disabled={true}
      onClick={[Function]}
      type="button">
      Book
    </button>
  </div>
</div>
`;
